src/mpid/ch4/netmod/ofi/ofi_init.h

```diff
@@ -649,6 +649,38 @@ static inline int MPIDI_NM_gpid_get(MPIR_Comm * comm_ptr, int rank, MPIR_Gpid *
     goto fn_exit;
 }
 
+static inline int MPIDI_NM_get_local_upids(MPIR_Comm *comm, size_t **local_upid_size,
+                                           char **local_upids)
+{
+    int mpi_errno = MPI_SUCCESS;
+    int i, total_size = 0;
+    char *temp_buf = NULL, *curr_ptr = NULL;
+
+    *local_upid_size = (size_t*) MPL_malloc(comm->local_size * sizeof(size_t));
+    temp_buf = (char*) MPL_malloc(comm->local_size * MPIDI_Global.addrnamelen);
+
+    for (i = 0; i < comm->local_size; i++) {
+        (*local_upid_size)[i] = MPIDI_Global.addrnamelen;
+        MPIDI_OFI_CALL(fi_av_lookup(MPIDI_Global.av, MPIDI_OFI_COMM_TO_PHYS(comm, i),
+                                            &temp_buf[i * MPIDI_Global.addrnamelen],
+                                            &(*local_upid_size)[i]), avlookup);
+        total_size += (*local_upid_size)[i];
```

Comment:
Yes. This is use to handle the mixed address sizes. I can adjust the ADI a little bit so that OFI can skip this if it knows the address size is fixed.